templates/commands/render.go

```diff
@@ -323,6 +347,72 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns them in a slice.
+func (r *Render) processUnknownInputs(spec *model.Spec) []string {
+	unknownInputs := make([]string, 0)
```

Comment:
I'd move this down after you've compiled `specInputs`, then you can efficiently allocate once:

```go
unknownInputs := make([]string, 0, len(r.flagInputs))
```

You might over-allocate, but you're guaranteed to only allocate once.